lib/Service/StackService.php

```diff
@@ -362,4 +368,93 @@ public function reorder($id, $order) {
 
 		return $result;
 	}
+
+	/**
+	 * @param $id
+	 * @param $boardId
+	 * @return Stack
+	 * @throws StatusException
+	 * @throws BadRequestException
+	 */
+	public function clone($id, $boardId) {
+		if (is_numeric($id) === false) {
+			throw new BadRequestException('stack id must be a number');
+		}
+
+		$this->permissionService->checkPermission(null, $boardId, Acl::PERMISSION_MANAGE);
+		if ($this->boardService->isArchived(null, $boardId)) {
+			throw new StatusException('Operation not allowed. This board is archived.');
+		}
+
+		$stack = $this->stackMapper->find($id);
+
+		$newStack = new Stack();
+		$newStack->setTitle($stack->getTitle() . ' (' . $this->l10n->t('copy') . ')');
+		$newStack->setBoardId($boardId);
+		$newStack->setOrder($stack->getOrder() +1);
```

Comment:
Ideally we get the number of stacks in total for the board and move it to the end.

Reply:
OK. This way I'll get the corresponding board:
$board = $this->boardMapper->find($boardId);
How can I get the stack number of this board?

Reply:
Feel free to set the order to 999 which is what we do for new stacks and cards anyways. They will then end up at the end of the stack list sorted by their creation time as a second sorting parameter.